.changeset/weak-swans-care.md

```diff
@@ -0,0 +1,5 @@
+---
+'xstate': minor
+---
+
+The event type of internal `after` events changed from `xstate.after(1000)#some.state.id` to `xstate.after.1000.#some.state.id` for consistency.
```

Comment:
One cool thing about the previous format was that "segments" were delimited in a somewhat better way. Theoretically one could define this:
```ts
setup({
  delays: {
    'wow.this.is.a.delay': () => 100
  }
})
```

With smth like this, we'll have a lot of segments delimited just by dots and it might be harder to tell which exact segments are coming from the `delayRef`. It's not exactly impossible using the proposed format but the previous format was more intuitive in this regard. OTOH, one could also define a stupid `')FU'` delay and mess up the matcher that would try to look between the opening and closing parenthesis. So perhaps it's not a big deal after all? Hm...

Please also note that even with the proposed adjustment we still can have some "special characters" in those generated event types:
https://github.com/statelyai/xstate/blob/63d923857592437dc174518ba02e061082f629cf/packages/core/src/stateUtils.ts#L295-297

It feels like this PR should adjust both or leave both alone.

Reply:
The small edge-case benefit of `xstate.after.*` is the partial event descriptor, for very advanced use-cases. The previous syntax was just me not knowing what I was doing.

I'll adjust our other magic IDs.